ocp_resources/resource.py

```diff
@@ -461,9 +463,19 @@ def _set_logger(self) -> logging.Logger:
 
     def _prepare_node_selector_spec(self) -> Dict[str, str]:
         if self.node_selector:
-            return {f"{self.ApiGroup.KUBERNETES_IO}/hostname": self.node_selector}
+            if isinstance(self.node_selector, dict):
+                return self.node_selector
+            else:
+                warn(
+                    "node_selector `str` will be deprecated in next release, pass `dict` instead",
```

Comment:
main would always be 10.*, if I am not mistaken. It would be good to clarify which version you are talking about.

Reply:
ACK

Reply:
remove from main since it not a realest branch, will add this to 4.16 branch